src/torque-offset-estimator.cpp

```diff
@@ -0,0 +1,240 @@
+            const Eigen::VectorXd& IMU_acc = m_accelerometerSIN(iter);
+            const Eigen::VectorXd& tau = m_jointTorquesSIN(iter);
+
+            Eigen::VectorXd enc(m_model.nq);
+            joints_sot_to_urdf(sot_enc.tail(m_model.nv-6), enc.tail(m_model.nv-6));
+            //enc.tail(m_model.nv-6) = sot_enc.tail(m_model.nv-6);
+            base_sot_to_urdf(sot_enc.head<6>(), enc.head<7>());
+
+            //Get the transformation from ff(f) to torso (t) to IMU(i) frame:
+            // fMi = oMf^-1 * fMt * tMi
+            se3::forwardKinematics(m_model,*m_data,enc);
+            se3::SE3 fMi = m_data->oMi[ffIndex].inverse()*m_data->oMi[torsoIndex]*m_torso_X_imu;
+
+            //Move the IMU signal to the base frame.
+            //angularAcceleration is zero. Intermediate frame acc and velocities are zero
+            const dynamicgraph::Vector acc = fMi.rotation()*IMU_acc; //Torso Acceleration
+
+            //Deal with gravity predefined in robot model. Robot Z should be pointing upwards
+            m_model.gravity.linear() = acc;
+            
+            //Set fixed for DEBUG
+            //m_model.gravity.linear() = m_model.gravity981;
+
+            const Eigen::VectorXd& tau_rnea = se3::rnea(m_model, *m_data, enc,
+                                                         Eigen::VectorXd::Zero(m_model.nv),
+                                                         Eigen::VectorXd::Zero(m_model.nv));
+            const Eigen::VectorXd current_offset = tau - tau_rnea.tail(m_model.nv-6);
+            if(current_offset.array().abs().maxCoeff() >=epsilon) {
+              SEND_MSG("Too high torque offset estimated for iteration"+ i, MSG_TYPE_ERROR);
+              assert(false);
+            }
+            // encSignals.push_back(_enc);
+            // accSignals.push_back(_acc);
+            // gyrSignals.push_back(_gyr);
+            // tauSignals.push_back(_tau);
+            jointTorqueOffsets += current_offset;
+            current_progress++;
+          }
+          else if (i == n_iterations){
+            // Find the mean, enough signals collected
+            jointTorqueOffsets /=n_iterations;
+            sensor_offset_status = COMPUTED;
+          }
+          else { // i > n_iterations
+            //Doesn't reach here.
+            assert(false && "Error in collectSensorData. ");
+          }
+        }
+        //        else { // sensor_offset_status == COMPUTED
+        //        } 
+        return s; 
+      }
+
+      DEFINE_SIGNAL_OUT_FUNCTION(jointTorquesEstimated, dynamicgraph::Vector)
+      {
+        m_collectSensorDataSINNER(iter);
+
+        const Eigen::VectorXd& gyro = m_gyroscopeSIN(iter);
+        if(gyro.array().abs().maxCoeff() >=gyro_epsilon) {
+          SEND_MSG("Very High Angular Rotations.", MSG_TYPE_ERROR_STREAM);
+        }        
+
+        if (s.size() != m_model.nv-6) s.resize(m_model.nv-6);
+
+        if (sensor_offset_status == PRECOMPUTATION || sensor_offset_status == INPROGRESS) {
```

Comment:
Try to be coherent with your policy for parenthesis. Typically you write the open parenthesis in a new line, but here you do it on the same line.